PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/survey/CustomerSurveyManager.java

```diff
@@ -0,0 +1,190 @@
+            final int delay = response == CustomerSurveyResponse.ACCEPT ? TAKE_SURVEY_DELAY_BY_DAY : PUT_OFF_DELAY_BY_DAY;
+            final CustomerSurveyConfiguration.CustomerSurveyStatus surveyStatus = getSurveyStatus(survey);
+            surveyStatus.setSurveyTimes(surveyStatus.getSurveyTimes() + 1);
+            surveyStatus.setLastSurveyDate(LocalDateTime.now());
+            surveyStatus.setNextSurveyDate(LocalDateTime.now().plusDays(delay * 2));
+            customerSurveyConfiguration.getSurveyStatus().stream()
+                    .filter(status -> !StringUtils.equalsIgnoreCase(status.getType(), survey.getType()))
+                    .forEach(customerSurveyStatus -> customerSurveyStatus.setNextSurveyDate(LocalDateTime.now().plusDays(delay)));
+        }
+        saveSurveyStatus();
+    }
+
+    private void saveSurveyStatus() {
+        final ObjectMapper mapper = new ObjectMapper();
+        final ObjectWriter writer = mapper.writer(new DefaultPrettyPrinter());
+        final File file = getConfigurationFile();
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            writer.writeValue(file, customerSurveyConfiguration);
+        } catch (final IOException e) {
+            e.printStackTrace();
+            // swallow exceptions for survey
+        }
+    }
+
+    private static File getConfigurationFile() {
+        return new File(BASE_FOLDER, FILE_NAME_SURVEY_CONFIG);
+    }
+
+    private static final class HOLDER {
+        private static final CustomerSurveyManager INSTANCE;
+
+        static {
+            CustomerSurveyConfiguration surveyConfiguration;
+            final ObjectMapper mapper = new ObjectMapper();
+            final File configurationFile = getConfigurationFile();
+            try {
+                surveyConfiguration = mapper.readValue(configurationFile, CustomerSurveyConfiguration.class);
+            } catch (final JsonProcessingException jsonException) {
+                try {
+                    final SurveyConfig config = mapper.readValue(configurationFile, SurveyConfig.class);
+                    final CustomerSurveyConfiguration.CustomerSurveyStatus surveyStatus = CustomerSurveyConfiguration.CustomerSurveyStatus.builder()
+                            .type(CustomerSurvey.AZURE_INTELLIJ_TOOLKIT.name())
+                            .surveyTimes(config.surveyTimes)
+                            .lastSurveyDate(config.lastSurveyDate)
+                            .nextSurveyDate(config.nextSurveyDate).build();
+                    surveyConfiguration = new CustomerSurveyConfiguration();
+                    surveyConfiguration.getSurveyStatus().add(surveyStatus);
+                } catch (final IOException e) {
+                    surveyConfiguration = new CustomerSurveyConfiguration();
+                }
+            } catch (final IOException e) {
+                surveyConfiguration = new CustomerSurveyConfiguration();
+            }
```

Comment:
How about to merge these two try catch blocks as one?

Reply:
For the `JsonProcessingException` exception, it may caused by the schema updates, so fall back to old schema here, but for IOException, no need to do the fallback, just return a new one.